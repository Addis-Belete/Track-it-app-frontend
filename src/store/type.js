export const FETCH_PROGRESS_REQUEST = 'FETCH_PROGRESS_REQUEST';
export const FETCH_PROGRESS_SUCCESS = 'FETCH_PROGRESS_SUCCESS';
export const FETCH_PROGRESS_FAILURE = 'FETCH_PROGRESS_FAILURE';
