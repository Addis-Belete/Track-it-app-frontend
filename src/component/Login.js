/* eslint-disable consistent-return */
/* eslint-disable jsx-a11y/anchor-is-valid */
import React, { useState } from 'react';

import Form from 'react-bootstrap/Form';
import { Button } from 'react-bootstrap/lib/InputGroup';

const Login = () => {
  const [name, setName] = useState('');
  const validateForm = () => (name.length > 0);
  const hanldleSubmit = (e) => {
    e.preventDefault();
  };
  return (
    <div className="login">
      <Form onSubmit={hanldleSubmit}>
        <Form.Group size="lg" controlId="name">
          <Form.Label>User Name</Form.Label>
          <Form.Control
            autoFocus
            type="text"
            value={name}
            onChange={(e) => setName(e.target.value)}
          />
        </Form.Group>
        <Button block size="lg" type="submit" disabled={!validateForm()}>
          Login
        </Button>
      </Form>

    </div>

  );
};
export default Login;
