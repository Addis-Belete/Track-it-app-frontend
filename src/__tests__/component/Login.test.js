import React from 'react';
import { cleanup, render } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Login from '../../component/Login';
import store from '../../store/store';
import '@testing-library/jest-dom';

afterEach(cleanup);
describe('addMeasurment', () => {
  it('should render a Login page', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>

      </Provider>,

    );
    expect(
      asFragment(
        <Provider store={store}>
          <Login />
        </Provider>,

      ),
    ).toMatchSnapshot();
  });
});
