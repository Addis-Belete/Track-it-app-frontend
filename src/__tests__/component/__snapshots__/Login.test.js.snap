// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addMeasurment should render a Login page 1`] = `
<DocumentFragment>
  <div
    class="login"
  >
    <input
      class="input"
      id="name"
      type="text"
      value=""
    />
    <button
      class="button"
      disabled=""
      type="button"
    >
      Login
    </button>
  </div>
</DocumentFragment>
`;
